SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestInitialOnly/SpatialTestInitialOnlyForSpawnActor.cpp

```diff
@@ -103,36 +104,24 @@ void ASpatialTestInitialOnlyForSpawnActor::PrepareTest()
 	});
 
 	AddStep(
-		TEXT("Check changed value."), FWorkerDefinition::Client(1),
-		[this]() -> bool {
-			bool IsReady = false;
-			TArray<AActor*> SpawnActors;
-			UGameplayStatics::GetAllActorsOfClass(GetWorld(), ASpatialTestInitialOnlySpawnActor::StaticClass(), SpawnActors);
-			for (AActor* Actor : SpawnActors)
-			{
-				ASpatialTestInitialOnlySpawnActor* SpawnActor = Cast<ASpatialTestInitialOnlySpawnActor>(Actor);
-				if (SpawnActor != nullptr)
-				{
-					IsReady = true;
-				}
-			}
-			return IsReady;
-		},
-		[this]() {
+		TEXT("Check changed value."), FWorkerDefinition::Client(1), nullptr, nullptr,
+		[this](float DeltaTime) {
```

Comment:
Why we use a `TickEvent` here instead?

Reply:
See my reply to Mike's comment here: https://github.com/spatialos/UnrealGDK/pull/2971/files#r595868053
This is the same kind of thing. If we set a replicated variable in the previous step on server 1, our usual flow for checking whether it has replicated to client 1 in this step would be to have a ticking step which checks, every tick, whether the information has arrived. That is what the `Require`s do - they test the condition supplied to see if it's true, but don't immediately fail if it's not true, the step just can't pass until all requires are satisfied. So in this case, the step will not finish until both Int_Initial == 1 and Int_Replicate == 2. Of course, in case this is not working, the step will never finish. This is why we usually use the timeout together with the requires. When the step times out, the requires will automatically provide debugging info for why the test failed, e.g.: "Check Actor.Int_Initial value. Failed, Received: 2, Expected: 1."

Reply:
And in this case, it also allows us to collapse the ready and start functions into one.

Reply:
`RequireEqual` and `FinishStep`, cool. 
I got it. Thanks.